qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -137,99 +142,200 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
+            one_zero = qubits[0] in neighbors1
             if zero_one and not one_zero:
                 preferred_direction = [0, 1]
             if one_zero and not zero_one:
                 preferred_direction = [1, 0]
         if (
-            self._natural_direction in {None, True}
+            natural_direction in {None, True}
             and preferred_direction is None
-            and layout
-            and self._backend_props
+            and gate_lengths
+            and gate_errors
         ):
             len_0_1 = inf
             len_1_0 = inf
-            try:
-                len_0_1 = self._backend_props.gate_length(
-                    decomposer2q.gate.name,
-                    [dag_qubit_index[node.qargs[0]], dag_qubit_index[node.qargs[1]]],
-                )
-            except BackendPropertyError:
-                pass
-            try:
-                len_1_0 = self._backend_props.gate_length(
-                    decomposer2q.gate.name,
-                    [dag_qubit_index[node.qargs[1]], dag_qubit_index[node.qargs[0]]],
-                )
-            except BackendPropertyError:
-                pass
-
-            if len_0_1 < len_1_0:
-                preferred_direction = [0, 1]
-            elif len_1_0 < len_0_1:
-                preferred_direction = [1, 0]
-            if preferred_direction:
-                physical_gate_fidelity = 1 - self._backend_props.gate_error(
-                    "cx", [dag_qubit_index[node.qargs[i]] for i in preferred_direction]
-                )
-        if self._natural_direction is True and preferred_direction is None:
+            twoq_gate_lengths = gate_lengths.get(decomposer2q.gate.name)
+            if twoq_gate_lengths:
+                len_0_1 = twoq_gate_lengths.get((qubits[0], qubits[1]), inf)
+                len_1_0 = twoq_gate_lengths.get((qubits[1], qubits[0]), inf)
+                if len_0_1 < len_1_0:
+                    preferred_direction = [0, 1]
+                elif len_1_0 < len_0_1:
+                    preferred_direction = [1, 0]
+                if preferred_direction:
+                    twoq_gate_errors = gate_errors.get("cx")
+                    gate_error = twoq_gate_errors.get(
+                        (qubits[preferred_direction[0]], qubits[preferred_direction[1]])
+                    )
+                    if gate_error:
+                        physical_gate_fidelity = 1 - gate_error
+        if natural_direction is True and preferred_direction is None:
             raise TranspilerError(
-                f"No preferred direction of {node.name} gate "
+                f"No preferred direction of gate on qubits f{qubits}"
                 "could be determined from coupling map or "
                 "gate lengths."
             )
-        basis_fidelity = self._approximation_degree or physical_gate_fidelity
-        su4_mat = node.op.to_matrix()
+        basis_fidelity = approximation_degree or physical_gate_fidelity
```

Comment:
0 is a valid value for `approximation_degree`, which is false-y.  Does this keep the desired behaviour if `approximation_degree = 0`?